link/src/main/java/com/stripe/android/link/ui/verification/VerificationViewModel.kt

```diff
@@ -0,0 +1,220 @@
+package com.stripe.android.link.ui.verification
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewModelScope
+import androidx.lifecycle.viewmodel.initializer
+import androidx.lifecycle.viewmodel.viewModelFactory
+import com.stripe.android.core.Logger
+import com.stripe.android.core.strings.ResolvableString
+import com.stripe.android.core.strings.resolvableString
+import com.stripe.android.link.LinkActivityResult
+import com.stripe.android.link.LinkScreen
+import com.stripe.android.link.account.LinkAccountManager
+import com.stripe.android.link.analytics.LinkEventsReporter
+import com.stripe.android.link.injection.NativeLinkComponent
+import com.stripe.android.link.model.AccountStatus
+import com.stripe.android.link.ui.ErrorMessage
+import com.stripe.android.link.ui.getErrorMessage
+import com.stripe.android.ui.core.elements.OTPSpec
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+/**
+ * ViewModel that handles user verification confirmation logic.
+ */
+internal class VerificationViewModel @Inject constructor(
+    private val linkAccountManager: LinkAccountManager,
+    private val linkEventsReporter: LinkEventsReporter,
+    private val logger: Logger,
+    private val goBack: () -> Unit,
+    private val navigateAndClearStack: (route: LinkScreen) -> Unit,
+    private val dismissWithResult: (LinkActivityResult) -> Unit
+) : ViewModel() {
+
+    private val _viewState = MutableStateFlow(
+        value = run {
+            val linkAccount = linkAccountManager.linkAccount.value
+            if (linkAccount == null) {
+                dismissWithResult(LinkActivityResult.Failed(NoLinkAccountFoundForVerification()))
+            }
+            VerificationViewState(
+                redactedPhoneNumber = linkAccount?.redactedPhoneNumber ?: "",
+                email = linkAccount?.email ?: ""
+            )
+        }
+    )
+    val viewState: StateFlow<VerificationViewState> = _viewState
+
+    val otpElement = OTPSpec.transform()
+
+    private val otpCode: StateFlow<String?> =
+        otpElement.otpCompleteFlow.stateIn(viewModelScope, SharingStarted.Lazily, null)
+
+    init {
+        setUp()
+    }
+
+    private fun setUp() {
+        val linkAccount = linkAccountManager.linkAccount.value
+
+        if (linkAccount == null) {
+            dismissWithResult(LinkActivityResult.Failed(NoLinkAccountFoundForVerification()))
+            return
```

Comment:
instead of requiring this check in multiple places, is it possible to pass the linkAccount into the VM or something, so we can ensure we only get to this screen if the link account exists?

Reply:
Makes sense, I changed it to have the LinkAccount in the constructor